include/nbl/asset/utils/IGLSLCompiler.h

```diff
@@ -22,17 +22,35 @@ namespace nbl::asset
 //! Will be derivative of IShaderGenerator, but we have to establish interface first
 class IGLSLCompiler final : public core::IReferenceCounted
 {
-		core::smart_refctd_ptr<IIncludeHandler> m_inclHandler;
-		system::ISystem* m_system;
-
 	public:
-		IGLSLCompiler(system::ISystem* _s);
+		enum E_SPIRV_VERSION
+		{
+			ESV_1_0 = 0x010000u,
+			ESV_1_1 = 0x010100u,
+			ESV_1_2 = 0x010200u,
+			ESV_1_3 = 0x010300u,
+			ESV_1_4 = 0x010400u,
+			ESV_1_5 = 0x010500u,
+			ESV_COUNT = 0xFFFFFFFFu
+		};
+
+		IGLSLCompiler(system::ISystem* _s, const E_SPIRV_VERSION targetSpirvVersion = ESV_1_3);
 
 		IIncludeHandler* getIncludeHandler() { return m_inclHandler.get(); }
 		const IIncludeHandler* getIncludeHandler() const { return m_inclHandler.get(); }
 
 		core::smart_refctd_ptr<ICPUBuffer> compileSPIRVFromGLSL(const char* _glslCode, IShader::E_SHADER_STAGE _stage, const char* _entryPoint, const char* _compilationId, bool _genDebugInfo = true, std::string* _outAssembly = nullptr, system::logger_opt_ptr logger = nullptr) const;
 
+		inline E_SPIRV_VERSION getTargetSpirvVersion() const
+		{
+			return m_targetSpirvVersion;
+		}
+
+		inline void setTargetSpirvVersion(const E_SPIRV_VERSION targetSpirvVersion)
+		{
+			m_targetSpirvVersion = targetSpirvVersion;
+		}
```

Comment:
why do we need to make target SPIRV version a global state of the compiler?

Couldn't we pass the desired target version with each compilation? (since `shaderc::CompileOptions` always asks for it)

It doesn;t seem to be some sort of a heavy thing thats costly mutate.

Reply:
I thought it would be "set once and forget about it" kinda thing, but sure I can make the methods take in a target spirv version

Reply:
Resolved